schedule/__init__.py

```diff
@@ -75,9 +75,17 @@ def run_all(self, delay_seconds=0):
             self._run_job(job)
             time.sleep(delay_seconds)
 
-    def clear(self):
-        """Deletes all scheduled jobs."""
-        del self.jobs[:]
+    def clear(self, tag=None):
+        """Deletes scheduled jobs marked with the given tag, or all jobs
+        if tag is omitted"""
+        if tag is None:
+            del self.jobs[:]
+        else:
+            old_ref = self.jobs
+            self.jobs = list([job for job in self.jobs if tag not in job.tags])
+            global jobs
+            if jobs is old_ref:
+                jobs = self.jobs
```

Comment:
I'm really starting to hate myself for these module level globals... ;D But I guess for now we'll have to deal with it and patch things up like you're doing here.